tutorial/part-2-using-forms/creating-a-form.md

```diff
@@ -0,0 +1,157 @@
+
+Let’s tinker with **ProductNameInput**:
+
+1. Open its properties
+2. Set **Data Type** to **Text** 
+3. Toggle **Required** field to Green
+
+The **Data Type** property sets the allowed data-type of input. When set to **Text**, it allows only **Text** input. The **Required** field makes the input to the field mandatory. You can toggle between true and false. The input widget has [many other properties available](https://docs.appsmith.com/widget-reference/input#properties).
+
+## Using a dropdown widget
+
+Let’s add other fields to the form. To enable users to choose a Category:
+
+1. Drag-drop a [dropdown widget](https://docs.appsmith.com/widget-reference/dropdown)
+2. Rename it to **CategoryDropdown**
+3. Set its **Select** **Type** to **Single Select**
+4. Add a [text widget](https://docs.appsmith.com/widget-reference/text) next to **CategoryDropdown** 
+5. Set its **Text** property to **Category:**
+
+By setting the **Selection Type** to **Single Select**, you allow the user to select only one value from the dropdown. You can alternatively configure a multi-select dropdown by choosing Multi Select from Selection Type. However, to do this tutorial, choose **Single Select** only.
+
+Note the **Categorydropdown**. It’s showing the options as - Vegetarian, Non-Vegetarian, and Vegan. These are its default options. Let’s change them to what's relevant to Oakry, that is - Vegetables, Bakery, Fruits, Beverages, and Dairy.
+
+1. Open **CategoryDropdown**’s properties modal.
+2. Place your cursor on its **Options** field. The value in this field is used to specify the options of the dropdown.
+3. Note the floating window show up when you place the cursor in **Options** field.
+4. See that the **Expected Data Type** is **Array&lt;{ label: string, value: string }&gt;**
   . You'll need to specify the options in the same format. 
+5. Set the **Options** field to that shown below.
+6. Set the **Default Option** to blank. This will keep the default selected option to blank.
+7. Toggle the **Required** field to true. This makes the input to the dropdown a required field.
+
+```text
+[
+  {
+    "label": "VEGETABLES",
+    "value": "VEGETABLES"
+  },
+  {
+    "label": "BAKERY",
+    "value": "BAKERY"
+  },
+  {
+    "label": "FRUITS",
+    "value": "FRUITS"
+  },
+	{
+    "label": "BEVERAGES",
+    "value": "BEVERAGES"
+  },
+  {
+    "label": "DAIRY",
+    "value": "DAIRY"
+  }
+]
+```
+
+## Writing your first transformation
+
+Notice that to set **CategoryDropdown**'s options, you had to copy-paste the values in a text editor. It was cumbersome. You can use JavaScript to simplify that:
+
+1. Open **CategoryDropdown**’s properties
+2. Copy-paste the following in its **Options** field: `{ { _.map(['vegetables', 'bakery', 'fruits', 'beverates', 'dairy'], function (category) { return { label: category.toUpperCase(), value: category.toUpperCase() } }) } }`
```

Comment:
Hmm... can't disagree there. But in this specific case, fetching from the DB doesn't make sense, because that limits the options to existing values in the DB. Thoughts?